From ce06a252d60e41fc41134a6ea7914e7645029ab2 Mon Sep 17 00:00:00 2001
From: Kai-Heng Feng <kai.heng.feng@canonical.com>
Date: Thu, 20 Aug 2020 22:12:02 +0800
Subject: [PATCH 50/51] Add support for Lenovo ThinkStation P620 Main Audio

This add support for its internal speaker and front headset.

Signed-off-by: Kai-Heng Feng <kai.heng.feng@canonical.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 .../Lenovo-ThinkStation-P620-Main-HiFi.conf        | 29 ++++++++++++++++++++++
 ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main.conf  |  6 +++++
 2 files changed, 35 insertions(+)
 create mode 100644 ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main-HiFi.conf
 create mode 100644 ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main.conf

diff --git a/ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main-HiFi.conf b/ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main-HiFi.conf
new file mode 100644
index 000000000000..c515d46699c7
--- /dev/null
+++ b/ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main-HiFi.conf
@@ -0,0 +1,29 @@
+SectionDevice."Headphones" {
+	Comment "Headphones"
+
+	Value {
+		PlaybackPriority 200
+		PlaybackPCM "hw:${CardId},1"
+		JackControl "Headphone - Output Jack"
+	}
+}
+
+SectionDevice."Mic" {
+	Comment "Mic"
+
+	Value {
+		CapturePriority 200
+		CapturePCM "hw:${CardId}"
+		JackControl "Mic - Input Jack"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+		JackControl "Speaker - Output Jack"
+	}
+}
diff --git a/ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main.conf b/ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main.conf
new file mode 100644
index 000000000000..53ef729acf03
--- /dev/null
+++ b/ucm2/USB-Audio/Lenovo-ThinkStation-P620-Main.conf
@@ -0,0 +1,6 @@
+Syntax 2
+Comment "USB-audio on Lenovo ThinkStation P620 Main Audio"
+SectionUseCase."HiFi" {
+	Comment "Default"
+	File "Lenovo-ThinkStation-P620-Main-HiFi.conf"
+}
-- 
2.16.4

